### PR TITLE
 feat(core): #38 di 

### PR DESCRIPTION
**Pull request by artalar**

### CODE REVIEW

File: docs/guides/di.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export const API = Symbol('api')
+export const apiAtom = declareAtom(API, axios, () => [])
+
+/* SOME FEATURE CODE */
+
+export const recieveData = declareAction()
+export const requestData = declareAction(async (payload, store) => {
+  const data = await store.getState(apiAtom).get(url, payload)
+  store.dispach(recieveData(data))
```
Comment (belozer) [inline docs/guides/di.md]: `dispatch`

File: packages/core/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@reatom/core",
   "private": false,
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (belozer) [inline packages/core/package.json]: bracking - major
feat - minor
fix - patch

`"version": "1.1.0"`

File: packages/core/tests/index.test.ts [hunk 3]
```diff
@@ -472,6 +472,21 @@ describe('@reatom/core', () => {
...
+    class Api {}
+    const api = new Api()
+    const apiAtom = declareAtom(Symbol('API'), api, () => [])
+
+    var store = createStore(apiAtom)
+    expect(store.getState()).toEqual({
+      [getTree(apiAtom).id]: api,
+    })
+
+    var store = createStore({ [getTree(apiAtom).id]: api })
```
Comment (belozer) [inline packages/core/tests/index.test.ts]: This is test about override.

```
const api2 = new Api()
var store = createStore({ [getTree(apiAtom).id]: api2 })
expect(store.getState(apiAtom)).toBe(api2)
```

**Comment by artalar** [comment]
@tsarapoid please, check this out #277

**Pull request closed without merging**